classes/command.js

```diff
@@ -1149,107 +1152,159 @@ class Command extends require("./template.js") {
+	 * @param {Array<string>} argsArray The arguments to parse from
+	 */
+	static parseParametersFromArguments (paramsDefinition, argsArray) {
+		const argsStr = argsArray.join(" ");
+		const outputArguments = [];
+		let parameters = {};
+
+		// Buffer used to store read characters before we know what to do with them
+		let buffer = "";
+		/** Parameter definition of the current parameter @type {typeof paramsDefinition[0] | null} */
+		let currentParam = null;
+		// is true if currently reading inside of a parameter
+		let insideParam = false;
+		// is true if the current param started using quotes
+		let quotedParam = false;
+
+		for (let i = 0; i < argsStr.length; i++) {
+			const char = argsStr[i];
+			buffer += char;
+
+			if (!insideParam) {
+				if (buffer.slice(0, -1) === Command.ignoreParametersDelimiter && char === " ") {
+					// Delimiter means all arguments after this point will be ignored, and just passed as-is
+					outputArguments.push(...argsStr.slice(i + 1).split(" "));
 					return {
-						success: false,
-						reply: `Cannot parse parameter "${name}"!`
+						success: true,
+						parameters,
+						args: outputArguments
 					};
 				}
 
-				if (type === "object") {
-					if (typeof parameters[name] === "undefined") {
-						parameters[name] = {};
-					}
-
-					if (typeof parameters[name][parsedValue.key] !== "undefined") {
-						return {
-							success: false,
-							reply: `Cannot use multiple values for parameter "${name}", key ${parsedValue.key}!`
-						};
+				if (char === ":") {
+					currentParam = paramsDefinition.find(i => i.name === buffer.slice(0,-1)) ?? null;
+					if (currentParam) {
+						insideParam = true;
+						buffer = "";
+						if (argsStr[i + 1] === "\"") {
+							i++;
+							quotedParam = true;
+						}
 					}
-
-					parameters[name][parsedValue.key] = parsedValue.value;
 				}
-				else {
-					parameters[name] = parsedValue;
+				else if (char === " ") {
+					const sliced = buffer.slice(0, -1);
+					if (sliced.length > 0) {
+						outputArguments.push(sliced);
+					}
+					buffer = "";
 				}
 			}
-		}
-
-		const remainingArgs = argsString.split(" ");
-		const paramRegex = new RegExp(`^(?<name>${paramNames.join("|")}):(?<value>.*)$`);
-		for (let i = remainingArgs.length - 1; i >= 0; i--) {
-			if (!paramRegex.test(remainingArgs[i])) {
-				continue;
-			}
 
-			const { name = null, value = null } = remainingArgs[i].match(paramRegex).groups;
-			const { type } = paramsDefinition.find(i => i.name === name);
-
-			if (name !== null && value !== null) {
-				const parsedValue = Command.parseParameter(value, type, false);
-				if (parsedValue === null) {
-					return {
-						success: false,
-						reply: `Cannot parse parameter "${name}"!`
-					};
+			if (insideParam) {
+				if (!quotedParam && char === " ") {
+					// end of unquoted param
+					const value = this.#parseAndAppendParameter(buffer.slice(0, -1), currentParam, quotedParam, parameters);
```

Comment:
I prefer using `Command.#parse...` instead of `this.#parse` - functionally it is the same, but it should be standardized across the file. And we already use `Command.parseParameter`, so just keep it that way.